pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -144,20 +141,34 @@ private QueryContext(@Nullable String tableName, @Nullable QueryContext subquery
     _explain = explain;
   }
 
+  public DataSourceContext getDataSource() {
+    return _dataSource;
+  }
+
   /**
    * Returns the table name.
-   * NOTE: on the broker side, table name might be {@code null} when subquery is available.
    */
   public String getTableName() {
-    return _tableName;
+    if (_dataSource.getSubquery() != null) {
+      return _dataSource.getSubquery().getTableName();
```

Comment:
So this handles the Standard SQL subquery where the inner query refers to the same table as outer query ?

How will this handle the generic subquery support (in multi stage engine) or is this specific to `IN_SUBQUERY` support in existing engine ?